src/plugins/dashboard/public/dashboard_container/embeddable/api/run_save_functions.tsx

```diff
@@ -222,86 +109,206 @@ export async function runQuickSave(this: DashboardContainer) {
   return saveResult;
 }
 
-export async function runClone(this: DashboardContainer) {
+/**
+ * @description exclusively for user directed dashboard save actions in both interaction modes; 'view' and 'edit', also
+ * accounts for scenarios of cloning elastic managed dashboard into user managed dashboards
+ */
+export async function runInteractiveSave(
+  this: DashboardContainer,
+  interactionMode: Extract<ViewMode, ViewMode.EDIT | ViewMode.VIEW>
```

Comment:
nit: if we know it's always either view or edit, why not just take a boolean argument here rather than extracting these values from the enum?

Reply:
I'd opted for this so the interaction mode is explicit provided, especially that this method is now generic enough to be called in any interaction (view) mode